rhp/v2/rhp.go

```diff
@@ -287,7 +269,7 @@ func (sh *SessionHandler) LocalAddr() string {
 }
 
 // NewSessionHandler creates a new RHP2 SessionHandler
-func NewSessionHandler(l net.Listener, hostKey types.PrivateKey, rhp3Addr string, cm ChainManager, s Syncer, wallet Wallet, contracts ContractManager, settings SettingsReporter, storage StorageManager, opts ...SessionHandlerOption) (*SessionHandler, error) {
+func NewSessionHandler(l net.Listener, hostKey types.PrivateKey, rhp3Addr string, cm ChainManager, s Syncer, wallet Wallet, contracts ContractManager, settings SettingsReporter, storage StorageManager, log *zap.Logger) (*SessionHandler, error) {
```

Comment:
Now that options are gone you could ask for the siamux port directly in the parameters and get rid of the error. The `NewSessionHandler` in rhp3 never returns an error so we could also remove it there. Bit of a nit but worth it in my opinion because you make it very clear why an RHP2 session handler needs the RHP3 address and you simplify the setup a bit.

Reply:
Done in #504